contracts/farm/SophonFarmingL2.sol

```diff
@@ -0,0 +1,756 @@
+
+    /// @notice Emitted when a user deposits to a pool
+    event Deposit(address indexed user, uint256 indexed pid, uint256 depositAmount, uint256 boostAmount);
+
+    /// @notice Emitted when a user withdraws from a pool
+    event Withdraw(address indexed user, uint256 indexed pid, uint256 amount);
+
+    /// @notice Emitted when a whitelisted admin transfers points from one user to another
+    event TransferPoints(address indexed sender, address indexed receiver, uint256 indexed pid, uint256 amount);
+
+    /// @notice Emitted when a user increases the boost of an existing deposit
+    event IncreaseBoost(address indexed user, uint256 indexed pid, uint256 boostAmount);
+
+    /// @notice Emitted when the the updatePool function is called
+    event PoolUpdated(uint256 indexed pid);
+
+    /// @notice Emitted when setPointsPerBlock is called
+    event SetPointsPerBlock(uint256 oldValue, uint256 newValue);
+
+    error ZeroAddress();
+    error PoolExists();
+    error PoolDoesNotExist();
+    error AlreadyInitialized();
+    error NotFound(address lpToken);
+    error FarmingIsStarted();
+    error FarmingIsEnded();
+    error TransferNotAllowed();
+    error TransferTooHigh(uint256 maxAllowed);
+    error InvalidEndBlock();
+    error InvalidDeposit();
+    error InvalidBooster();
+    error InvalidPointsPerBlock();
+    error InvalidTransfer();
+    error WithdrawNotAllowed();
+    error WithdrawTooHigh(uint256 maxAllowed);
+    error WithdrawIsZero();
+    error NothingInPool();
+    error NoEthSent();
+    error BoostTooHigh(uint256 maxAllowed);
+    error BoostIsZero();
+    error BridgeInvalid();
+    error OnlyMerkle();
+
+    address public immutable MERKLE;
+
+    /**
+     * @notice Construct SophonFarming
+     */
+    constructor(address _MERKLE) {
+        MERKLE = _MERKLE;
+    }
+
+
+    // Order is important
+    function addPool(
+        uint256 _pid,
+        IERC20 _lpToken,
+        address _l2Farm,
+        uint256 _amount,
+        uint256 _boostAmount,
+        uint256 _depositAmount,
+        uint256 _allocPoint,
+        uint256 _lastRewardBlock,
+        uint256 _accPointsPerShare,
+        uint256 _totalRewards,
+        string memory _description,
+        uint256 _heldProceeds
+    ) public onlyOwner {
+        require(_amount == _boostAmount + _depositAmount, "balances don't match");
+        poolInfo[_pid] = PoolInfo({
+            lpToken: _lpToken,
+            l2Farm: _l2Farm,
+            amount: _amount,
+            boostAmount: _boostAmount,
+            depositAmount: _depositAmount,
+            allocPoint: _allocPoint,
+            lastRewardBlock: _lastRewardBlock,
+            accPointsPerShare: _accPointsPerShare,
+            totalRewards: _totalRewards,
+            description: _description
+        });
+        heldProceeds[_pid] = _heldProceeds;
+        require(IERC20(_lpToken).balanceOf(address(this)) >= _amount, "balances don't match");
```

Comment:
This line will revert. Instead use: require(IERC20(_lpToken).balanceOf(address(this)) >= _depositAmount, "balances don't match");